cadence/contracts/AxelarGateway.cdc

```diff
@@ -0,0 +1,344 @@
+      self.payloadHash = payloadHash
+      self.sourceTxHash = sourceTxHash
+      self.sourceEventIndex = sourceEventIndex
+    }
+  }
+
+  access(all) let executedCommandIds: [String]
+  access(self) let approvedCommands: @{String: CGPCommand}
+
+  init() {        
+    self.SELECTOR_TRANSFER_OPERATORSHIP = Crypto.hash("transferOperatorship".utf8, algorithm: HashAlgorithm.KECCAK_256)
+    self.SELECTOR_APPROVE_CONTRACT_CALL = Crypto.hash("approveContractCall".utf8, algorithm: HashAlgorithm.KECCAK_256)
+    self.PREFIX_APP_CAPABILITY_NAME = "AppCapabilityPath"
+
+    self.executedCommandIds = []
+    self.approvedCommands <- {}
+  }
+
+  access(all) struct ParamHandler {
+    access(all) let rawParams: [AnyStruct]
+
+    init(params: [AnyStruct]) {
+      self.rawParams = params
+    }
+
+    access(contract) fun generateApproveContractCallParams(): ApproveContractCallParams? {
+      let sourceChain = self.rawParams[0].isInstance(Type<String>())
+      let sourceAddress = self.rawParams[1].isInstance(Type<String>())
+      let contractAddress = self.rawParams[2].isInstance(Type<String>())
+      let payloadHash = self.rawParams[3].isInstance(Type<String>())
+      let sourceTxHash = self.rawParams[4].isInstance(Type<String>())
+      let sourceEventIndex = self.rawParams[5].isInstance(Type<UInt256>())
+
+      if !sourceChain || !sourceAddress || !contractAddress || !payloadHash || !sourceTxHash || !sourceEventIndex {
+        return nil
+      }
+
+      return ApproveContractCallParams(
+        sourceChain : self.rawParams[0] as! String,
+        sourceAddress : self.rawParams[1] as! String,
+        contractAddress : self.rawParams[2] as! String,
+        payloadHash : self.rawParams[3] as! String,
+        sourceTxHash : self.rawParams[4] as! String,
+        sourceEventIndex : self.rawParams[5] as! UInt256,
+      )
+    }
+
+    access(contract) fun generateTransferOperatorshipParams(): AxelarAuthWeighted.TransferOperatorshipParams? {
+      let newOperators = self.rawParams[0].isInstance(Type<[String]>())
+      let newWeights = self.rawParams[1].isInstance(Type<[UInt256]>())
+      let newThreshold = self.rawParams[2].isInstance(Type<UInt256>())
+
+      if !newOperators || !newWeights || !newThreshold {
+        return nil
+      }
+
+      return AxelarAuthWeighted.TransferOperatorshipParams(
+        newOperators : self.rawParams[0] as! [String],
+        newWeights : self.rawParams[1] as! [UInt256],
+        newThreshold : self.rawParams[2] as! UInt256,
+      )
+    }
+  }
+
+  access(all) resource interface Executable {
+    access(all) fun executeApp(commandResource: &AxelarGateway.CGPCommand, sourceChain: String, sourceAddress: String, payload: [UInt8])
+  }
+
+  access(all) resource interface SenderIdentity {}
+
+  access(all) fun callContract(senderIdentity: Capability<&{AxelarGateway.SenderIdentity}>, destinationChain: String, destinationContractAddress: String, payload: [UInt8]) {
```

Comment:
Is there supposed to be a fee associated with this call? Wondering how these calls are funded.